opentelemetry-api/src/opentelemetry/resources/__init__.py

```diff
@@ -11,3 +11,56 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from typing import Any, Dict, Union
+
+
+class Resource:
+    def __init__(self, labels: Dict[str, str]):
```

Comment:
The implementation should be in the SDK package at `opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py`.

We included some implementation in the tracer API because we need `Tracer` to produce real (but invalid) spans if there's no SDK loaded. AFAIK that's not the case for resources, we may only need to use them from the SDK. This is not described in the spec, which doesn't distinguish much between the API and SDK.

That said there is some [real code in the java API](https://github.com/open-telemetry/opentelemetry-java/blob/0478651e34ede5960d6275626c97103e32470846/api/src/main/java/io/opentelemetry/resources/Resource.java), so I might be missing something here.

Reply:
moved to sdk.